### ISSUE TITLE
Use linkedin/coral for converting Spark expressions in Delta Lake connector

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#17134]
Currently, the connector uses ANTLR to convert Spark expressions to Trino expressions. 
We should consider switching to linkedin/coral once they support translating Spark SQL to RelNode. 

Relates to https://github.com/linkedin/coral/pull/339

**Comment by wmoustafa** [comment]
Thanks @ebyhr for filing this issue, and filing the respective Coral PR as well. Let us circle back by June when current Coral refactor is expected to be done. FYI @dain.

**Comment by chenjian2664** [comment]
Hi @ebyhr I am new for the delta-lake connector, one question related issue, why we need this transformation, and when will use it?

**Comment by ebyhr** [comment]
@chenjian2664 Databricks and OSS Delta Lake use Spark expressions for check constraints and generated columns. You can refer to https://github.com/delta-io/delta/blob/master/PROTOCOL.md. We need to convert them to Trino expressions because those syntaxes are different. For instance, Spark uses `&` for bitwise AND, but Trino uses `bitwise_and` function.

**Comment by chenjian2664** [comment]
@ebyhr Got it, thank you

**Comment by herry-heni** [comment]
@ebyhr is this issue related with this error:

```
Query 20230911_153258_00008_xmmwt failed: Unsupported Spark expression: length(name) > 0
```

My Delta table is using this check constraint:

```
'delta.constraints.nameValid' = 'length(name) > 0'
```

I am using Trino version `407`.

**Comment by ebyhr** [comment]
@herry-heni The failure should be fixed in this enhancement.

**Comment by anilsomisetty** [comment]
@ebyhr any update on this integration ?

**Comment by ebyhr** [comment]
I don't think we want to use the library as it's not maintained well in my opinion.

**Comment by wmoustafa** [comment]
@ebyhr Coral’s recent development has been focused on improving query compliance and testability—both of which are essential for maintaining any query compilation or translation engine. These improvements have been developed internally at LinkedIn first and haven’t yet been pushed to OSS. So while they’re not visible externally, there are in fact significant maintenance enhancements coming soon.

**Comment by wmoustafa** [comment]
To add to the above, those query compliance features would be a pre-requisite for the work required in this issue in my opinion.
